### DISCUSSION TITLE
Add one-line using without variable

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/csharplang#7882] [Language Ideas]
What I mean is instead of doing `using var _ = GIL.Acquire();` it would make sense to enable `using GIL.Acquire();`. This same feature is already enabled in block using:

```csharp
using (GIL.Acquire())
{
    // do stuff
}
```

Akin to `defer` in Golang and others, but this one will support only `IDisposable` and not just any statement. 
I think this will make the code simpler.

**Comment by CyrusNajmabadi** [comment]
The challenge here is that is `using expr;` is already legal syntax, and this would be ambiguous in a top level program.

**Reply by alexrp** [reply]
If there isn't a solution to the ambiguity, it seems like it might be reasonable to just say "tough luck, you can't do that, unless you move it into a block of some kind". Even as a user of top-level programs myself, I'd wager almost nobody would hit this restriction, and those few that do would have an easy alternative. Seems like a small price to pay. 🤔

**Comment by RenderMichael** [comment]
I think he best syntax for this would be 
``` csharp
using _ = GIL.Acquire();
```

No conflicts with existing syntax (this doesn’t compile today including on top-level statements), and I think a discard is a very good indicator to the reader that a value is generated, and ignored.

**Reply by HaloFour** [reply]
It is still legal syntax, as `_` is still a valid identifier for aliases.  It would only fail to compile during semantic analysis when the expression doesn't yield a disposable value.

```cs
using _ = System.Console;
```

**Reply by RenderMichael** [reply]
I guess all we can hope for is for the breaking changes discussion to go through and for discards-as-identifiers to be obsoleted

**Comment by jnm2** [comment]
See existing discussions on this:

- https://github.com/dotnet/csharplang/discussions/2593 (another duplicate: https://github.com/dotnet/csharplang/discussions/6860)

- Championed proposal: https://github.com/dotnet/csharplang/issues/2235 

- `await using` use case for this, because `ConfigureAwait` does not return a usable object to assign to a variable:
  https://github.com/dotnet/csharplang/discussions/2661

**Reply by B1Z0N** [reply]
Thanks. Very helpful!
